### PR TITLE
Take location from state file for az jump host

### PR DESCRIPTION
**Pull request by derpsteb**
### Proposed change(s)

- I was hitting a resource not found error because I was using different locations. This patch should prevent that error from happening.

### CODE REVIEW

**Review by malt3 — DISMISSED** [review]
I'm wondering what the jump host can still be used for. I was considering removing it since SSH is not available anymore. If it is useful for your purposes we should keep it :+1: 

<strike>LGTM</strike>

EDIT: whoops. @Nirusu has a point.

File: hack/azure-jump-host/jump-host-create [hunk 1]
```diff
@@ -5,6 +5,7 @@ set -o pipefail
 SCRIPTDIR="$( dirname --  $(realpath "${BASH_SOURCE}"); )";
 RG=$(jq -r .azureresourcegroup constellation-state.json)
 SUBNET=$(jq -r .azuresubnet constellation-state.json)
+LOCATION=$(jq -r .azurelocation constellation-state.json)
```
Comment (Nirusu) [inline hack/azure-jump-host/jump-host-create:8]: What version have you tested this with? We don't have a state file on the current main branch?

Comment (derpsteb) [reply hack/azure-jump-host/jump-host-create:8]: I needed this to recover a v2.0 cluster. Maybe this should be on the v2.0 branch? The docs for v2.0 in the recovery section could also point to this hack folder. They currently only state that you should setup a jump host.

Comment (malt3) [reply hack/azure-jump-host/jump-host-create:8]: I think that's a good idea. Can you rebase this PR to target `release/v2.0`?

**Comment by derpsteb** [comment]
Moved the change to only happen on `release/v2.0`. 
Since this is no longer needed on main we can remove the script from main.

**Review by malt3 — APPROVED** [review]
LGTM

**Pull request merged**